src/KubernetesClient/StreamDemuxer.cs

```diff
@@ -199,6 +204,13 @@ protected async Task RunLoop(CancellationToken cancellationToken)
 
                     var streamIndex = buffer[0];
                     var extraByteCount = 1;
+                    byteCount += result.Count - extraByteCount;
+                    if (this.streamType == StreamType.PortForward && byteCount <= 2)
```

Comment:
I'm not sure this will work correctly, what if byteCount becomes greater than 2 immediately, you won't strip the first two bytes.

I think the right thing to do is to have a `boolean skippedPort` outside the loop and then clip the two bytes at the front if `!skippedPort` and then set `skippedPort` to true...

Eitherway, I think this deserves a unit test to make sure it's working.

Thanks!

Reply:
here's another pass. there was a mistake in my previous change where I didn't handle StreamDemuxer with multiple streams. Now there is a dictionary that holds the skipped bytes for each stream.
it turns out there is 0 StreamDemuxer tests. add StreamDemuxerTests that covers send / receive for both command and port forwarding. Added WebSocketMock as a helper.